lib/redlock/client.rb

```diff
@@ -199,6 +199,17 @@ def try_lock_instances(resource, ttl, options)
       false
     end
 
+    def attempt_retry_delay(attempt_number)
+      retry_delay =
+        if @retry_delay.respond_to?(:call)
```

Comment:
🤔I think it might be better for performance to check this only once and always have a proc - but not sure to be honest, the penalty for an unneeded proc call might be higher than this.

ie, something like:

```rb
# initialize
@retry_delay = options[:retry_delay] || DEFAULT_RETRY_DELAY
@retry_delay = ->(_attempt_number) { @retry_delay } unless @retry_delay.respond_to?(:call)
```

Reply:
As most of the time this argument will be a simple number, it should be cheaper to check for callable than calling unneeded proc.

Reply:
🤔 Maybe, but you're checking every time in `attempt_retry_delay` (so `@retry_count` times per failed lock, or potentially more often when people re-use the same `Redlock::Client` instance)